src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -194,7 +202,6 @@ public void setType(EntryType type) {
     /**
      * Returns the contents of the given field, or null if it is not set.
      */
-    @Deprecated //Use getFieldOptional instead
```

Comment:
As far as I can see it you need it for int and string, right? so just overload the method. Anyway, accepting a parameter of type `Object` smells bad.